gobblin-metrics-libs/gobblin-metrics/src/main/java/org/apache/gobblin/metrics/GobblinMetrics.java

```diff
@@ -367,6 +374,20 @@ public void startMetricReportingWithFileSuffix(State state, String metricsFileSu
       oldMetricsFileSuffix += "." + metricsFileSuffix;
     }
     metricsReportingProps.setProperty(ConfigurationKeys.METRICS_FILE_SUFFIX, oldMetricsFileSuffix);
+
+    if (!Strings.isNullOrEmpty(metricsFileSuffix)) {
+      String[] tokens = metricsFileSuffix.split("_");
```

Comment:
I just googled and found that there are mapreduce classes for these such as `TaskAttemptID`. Would it be possible to make use of these to get the properties through APIs instead of parsing the string manually?

Reply:
I could not find any API which gives task number and attempt. getTaskAttemptID().getJobID(), getTaskAttemptID().getTaskID(), getTaskAttemptID().getId() did not quite give what I am trying to extract from taskAttemptID.
I'll use context.getTaskAttemptID().getTaskType()